5.3.Dish_Share.py

```diff
@@ -0,0 +1,25 @@
+
+# define a generator that return the perfect numbers from 1 to infinity
+def perfect_numbers():
+    # define a function that get a number and check if the sum of the divisors of the number is equal to the number
+    # if the sum is equal to the number return True else return False
+    def is_perfect(num):
+        # create a list of the divisors of the number
+        divisors = [i for i in range(1, int(num / 2) + 1) if num % i == 0]
```

Comment:
Consider replacing 'i' with a more explanatory name, Variable names - https://workat.tech/machine-coding/tutorial/writing-meaningful-variable-names-clean-code-za4m83tiesy0 - variable names, abbreviation reference is missing